### ISSUE TITLE
Shots option from blackbird script is lost during `io.to_program()`

### ISSUE THREAD

**Issue by co9olguy** [XanaduAI/strawberryfields#129]
When loading a blackbird script from file and converting it to a program, `to_program` uses the `target` attribute of the `BlackbirdProgram` object during a compilation step, but seems to ignore other options. Unfortunately, this is where the `shots` parameter lives, thus this information never makes it to the final program object. During execution with `StarshipEngine`, it is replaced by the default `shots=1`.

**Comment by josh146** [comment]
On it.

**Comment by josh146** [comment]
I've looked through the code, what is appears to be happening is this:

1. `to_program` correctly passes all target options as `**kwargs` to `program.compile`.

2. Unfortunately, `program.compile` ignores all `**kwargs` options not to do with compilation, so at this point `shots` gets lost.

What I can do is modify `Program` to store a list of `run_options` for the compiled program, which can then be accessed by the `Engine`.

This will work, I guess the question is, is this a good idea? Do we want the `Program` to be able to specify `run_options` for later execution?

I remember this came up previously, and we decided we did not want this option.
